pkg/client/client.go

```diff
@@ -1431,7 +1437,16 @@ func (c *Client) WaitForDaemonSetRollout(ctx context.Context, ds *appsv1.DaemonS
 }
 
 func (c *Client) CreateOrUpdateSecret(ctx context.Context, s *v1.Secret) error {
-	sClient := c.kclient.CoreV1().Secrets(s.GetNamespace())
+	// TEMP: just to make sure we never touch secrets created by service-ca-operator.
+	if c.maybeHasServiceCAData(ctx, s) {
```

Comment:
@jan--f, I'd like your input on this.
I don't think CMO touches secrets created by `service-ca-operator` or should do so.
Do you remember for which cases the `maybeHasServiceCAData` check was introduced.
(A lot of things have changed since then :))

Reply:
I think you're right.
In https://github.com/openshift/cluster-monitoring-operator/pull/1495 we discussed this and there is a discussion that menions CMO adding/removing metadata, but the example linked doesn't do that for secrets. 
And I can't find code that touches (or even knows) about the service-ca-op created secrets.
@simonpasquier can you think of anything we're missing here?

Reply:
Ok, thanks to you two for the confirmation, I'll clean up the code then.
For the configmaps, https://github.com/openshift/library-go/pull/1676 should help CMO reduce unnecessary competition.
for ValidatingWebhookConfiguration, the existing ressourceapply should be sufficient.